lib/rubocop/cop/rails/active_record_bulk_persistence.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Checks for iterative save/create/update methods instead of using insert_all/upsert_all
+      #
+      # @example
+      #   # bad
+      #   events.each do |user|
+      #     event.save!
+      #   end
+      #
+      #   # good
+      #   Event.insert_all(events)
+      #
+      class ActiveRecordBulkPersistence < Base
+        include ActiveRecordHelper
+
+        MSG = 'For bulk operations, use `insert_all` or `upsert_all` instead of repeated calls to `%<method>s`.'
+
+        SCOPE_METHODS = %i[
+          all eager_load includes joins left_joins left_outer_joins not preload
+          references unscoped where
+        ].freeze
+        ITERATIVE_METHODS = %i[each find_each each_with_index each_with_object reduce inject map flat_map].freeze
```

Comment:
Is there some way to validate that this is comprehensive?

Reply:
done, thanks

Reply:
I added a couple of missing methods: filter_map, collect.
I based the list on enumerable [doc](https://ruby-doc.org/core-3.1.2/Enumerable.html).